partest/src/main/scala/scala/tools/partest/scalajs/ScalaJSPartest.scala

```diff
@@ -124,57 +124,41 @@ trait ScalaJSSuiteRunner extends SuiteRunner {
 
   override def runTestsForFiles(kindFiles: Array[File],
       kind: String): Array[TestState] = {
-    super.runTestsForFiles(kindFiles.filter(shouldUseTest), kind)
+    super.runTestsForFiles(kindFiles.filter(testFilter), kind)
   }
 
   private lazy val listDir =
     s"/scala/tools/partest/scalajs/$scalaVersion"
 
-  private lazy val buglistedTestFileNames =
-    readTestList(s"$listDir/BuglistedTests.txt")
+  private lazy val blacklistedTests = {
+    val source = scala.io.Source.fromURL(getClass.getResource(
+        s"$listDir/BlacklistedTests.txt"))
 
-  private lazy val blacklistedTestFileNames =
-    readTestList(s"$listDir/BlacklistedTests.txt")
-
-  private lazy val whitelistedTestFileNames =
-    readTestList(s"$listDir/WhitelistedTests.txt")
-
-  private def readTestList(resourceName: String): Set[String] = {
-    val source = scala.io.Source.fromURL(getClass.getResource(resourceName))
-
-    val fileNames = for {
+    val files = for {
       line <- source.getLines
       trimmed = line.trim
       if trimmed != "" && !trimmed.startsWith("#")
-    } yield extendShortTestName(trimmed)
+    } yield {
+      extendShortTestName(trimmed)
+    }
 
-    fileNames.toSet
+    files.toSet
   }
 
-  private def extendShortTestName(testName: String) = {
-    val srcDir = PathSettings.srcDir
-    (srcDir / testName).toCanonical.getAbsolutePath
+  private def extendShortTestName(testName: String): File = {
+    val f = PathSettings.srcDir / testName
+    require(f.exists(), s"$testName does not exist")
+    f
```

Comment:
Dropping `toCanonical.getAbsolutePath` in `extendShortTestName` also has an impact on the `SomeTests` test filter. Did you check that we can still do

    > partestSuite testOnly -- -- run/sometest.scala

(hopefully also on Windows)

Reply:
Yes, still works. In fact, we now get checking that the tests passed on the command line actually exist.